dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityFilterWriter.java

```diff
@@ -0,0 +1,56 @@
+package com.dotcms.rendering.velocity.servlet;
+
+import java.io.FilterWriter;
+import java.io.IOException;
+import java.io.Writer;
+
+class VelocityFilterWriter extends FilterWriter {
+
+    private boolean firstNonWhiteSpace = false;
+
+    public VelocityFilterWriter(Writer arg0) {
+        super(arg0);
+
+    }
+
+    @Override
+    public void write(char[] arg0) throws IOException {
+        if (firstNonWhiteSpace) {
+            super.write(arg0);
+        } else {
+
+            for (int i = 0; i < arg0.length; i++) {
+                if (arg0[i] > 32) {
+                    firstNonWhiteSpace = true;
+                }
+                if (firstNonWhiteSpace) {
+                    super.write(arg0[i]);
+                }
+
+            }
+
+        }
+
+    }
+
+    @Override
+    public void write(String arg0) throws IOException {
+        if (firstNonWhiteSpace) {
+            super.write(arg0);
+        } else {
+            char[] stringChar = arg0.toCharArray();
+            for (int i = 0; i < stringChar.length; i++) {
+
+                if (stringChar[i] > 32) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 32 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS109)